pkg/util/general/window.go

```diff
@@ -100,6 +115,20 @@ type sample struct {
 	timestamp time.Time
 }
 
+func NewAggregatorSmoothWindow(args SmoothWindowArgs) SmoothWindow {
+	if args.AggregatorName == SmoothWindowAggregatorAvg {
+		return NewAverageWithTTLSmoothWindow(args.WindowSize, args.TTL, args.UsedMillValue)
+	} else if strings.HasPrefix(args.AggregatorName, "p") || strings.HasPrefix(args.AggregatorName, "P") {
```

Comment:
please define a new AggregatorName rather than hacking with prefix "p", and change this implementation to `switch/case` rather than `if/else`